hub/core/storage/s3.py

```diff
@@ -2,16 +2,16 @@
 import botocore  # type: ignore
 import posixpath
 from typing import Optional
-from collections.abc import MutableMapping
+from hub.core.storage.provider import StorageProvider
 from hub.util.exceptions import S3GetError, S3SetError, S3DeletionError, S3ListError
 
 
-class S3Mapper(MutableMapping):
-    """An s3 mapper built using boto3. For internal use only (by class S3Provider)"""
+class S3Provider(StorageProvider):
+    """Provider class for using S3 storage."""
 
     def __init__(
         self,
-        url: str,
+        root: str,
```

Comment:
Can we allow the user to pass an s3Client. If no client then we use the given code to create a default client.